lib/Optimizer/Transforms/DecompositionPatterns.cpp

```diff
@@ -154,6 +172,23 @@ class QuakeOperatorCreator {
     return op;
   }
 
+  template <typename OpTy>
+  OpTy create(Location location, Type &measOut, SmallVector<Type> &wires,
+              SmallVector<Value> &targets, StringAttr &registerNameAttr) {
+    OpTy op;
+    op = rewriter.create<OpTy>(location, measOut, wires, targets,
+                               registerNameAttr);
+
+    auto resultWires = op.getWires();
+    auto resultIt = resultWires.begin();
+    auto resultWiresEnd = resultWires.end();
+    for (auto &t : targets)
+      if (t.getType().isa<quake::WireType>() && resultIt != resultWiresEnd)
+        t = *resultIt++;
```

Comment:
I'd rather use a local counter and dereference resultWires directly than pointer arithmetic
operator[] gives you some assertion protection